layout-server/src/test/java/org/finos/vuu/layoutserver/integration/ApplicationLayoutIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+package org.finos.vuu.layoutserver.integration;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.finos.vuu.layoutserver.exceptions.InternalServerErrorException;
+import org.finos.vuu.layoutserver.model.ApplicationLayout;
+import org.finos.vuu.layoutserver.repository.ApplicationLayoutRepository;
+import org.finos.vuu.layoutserver.utils.DefaultApplicationLayoutLoader;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+public class ApplicationLayoutIntegrationTest {
+    public static final ObjectMapper objectMapper = new ObjectMapper();
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ApplicationLayoutRepository repository;
+    @MockBean
+    private DefaultApplicationLayoutLoader mockLoader;
+    private final DefaultApplicationLayoutLoader realLoader = new DefaultApplicationLayoutLoader();
+
+    @Test
+    public void getApplicationLayout_noLayoutExists_returns200WithDefaultLayout() throws Exception {
```

Comment:
do we want to cover the scenario when the header doesn't have a user?

Reply:
Rolled up with a bit of refactor in here: d9b27dae